### DISCUSSION TITLE
Can you show spaces in code blocks?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#4791] [Q&A]
Have a page listing markdown examples. And because of that would it be nice if I could somehow display spaces in the code blocks (i.e. for a hard-wrap example).

Is this doable with CSS or similar?

**Comment by facelessuser** [comment]
Can you elaborate with an example?

**Reply by Andre601** [reply]
Here I have an example about line breaks (using two spaces for a hard wrap):
https://www.andre601.ch/GitHub-Markdown/github/general/#line-breaks

I would like to have the spaces being visible, i.e. with a centered dot, so that people know that there is something.

I recall the GFM reference page having such a thing for their examples...

**Reply by facelessuser** [reply]
You would have to apply special rules to ensure leading/trailing white space is meaningful as browsers normally would ignore it in the code block (`white-space: break-spaces;`). This would not be a Material request though, it would have to be an Python Markdown feature request. Maybe it could be done in InlineHilite. But we'd have to figure out a way not to break existing behavior.

Currently, InlineHilite, relies on the default behavior of white space at the beginning and end getting stripped. If I want to inline highlight some text, I can use `#!python import stuff`. If I want a literal `#!python`, I need to provide a space before it : ` #!python`.

I've thought about in the past that the first white space could be ignored, so to show one leading white space, you'd add two. It was just a thought. I think GitHub always respects literal white space at the start and end, but this is GitHub specific and not something that all CommonMark respects: https://babelmark.github.io/?text=%60++this++%60%0A%0A%60+this+%60%0A%0A%60this%60.

**Reply by Andre601** [reply]
Not sure, but I feel like you misunderstood me? Idk to be honest...

**Reply by facelessuser** [reply]
> I would like to have the spaces being visible, i.e. with a centered dot, so that people know that there is something.

I see this portion now, I overlooked it. Well, I guess you could use literal dots. Or you'd need some kind of post processor to wrap white spaces and style them differently, maybe inserting some indicator.
